lib/puppetserver/ca/create_action.rb

```diff
@@ -0,0 +1,268 @@
+require 'puppetserver/ca/utils'
+require 'puppetserver/ca/host'
+require 'puppetserver/ca/puppet_config'
+require 'puppetserver/utils/file_utilities'
+require 'puppetserver/utils/refactored_client'
+require 'puppetserver/utils/signing_digest'
+require 'json'
+
+module Puppetserver
+  module Ca
+    class CreateAction
+
+      include Puppetserver::Utils
+
+      # Only allow printing ascii characters, excluding /
+      VALID_CERTNAME = /\A[ -.0-~]+\Z/
+      CERTNAME_BLACKLIST = %w{--all --config}
+
+      SUMMARY = "Create a new certificate signed by the CA"
+      BANNER = <<-BANNER
+Usage:
+  puppetserver ca create [--help]
+  puppetserver ca create [--config] --certname CERTNAME[,ADDLCERTNAME]
+
+Description:
+Creates a new certificate signed by the intermediate CA
+and stores generated keys and certs on disk.
+
+To determine the target location, the default puppet.conf
+is consulted for custom values. If using a custom puppet.conf
+provide it with the --config flag
+
+Options:
+BANNER
+      def initialize(logger)
+        @logger = logger
+      end
+
+      def self.parser(parsed = {})
+        parsed['certnames'] = []
+        OptionParser.new do |opts|
+          opts.banner = BANNER
+          opts.on('--certname FOO,BAR', Array,
+               'One or more comma separated certnames') do |certs|
+            parsed['certnames'] += certs
+          end
+          opts.on('--help', 'Display this create specific help output') do |help|
+            parsed['help'] = true
+          end
+          opts.on('--config CONF', 'Path to puppet.conf') do |conf|
+            parsed['config'] = conf
+          end
+        end
+      end
+
+      def parse(args)
+        results = {}
+        parser = self.class.parser(results)
+
+        errors = Utils.parse_with_errors(parser, args)
+
+        if results['certnames'].empty?
+          errors << '    At least one certname is required to create'
+        else
+          results['certnames'].each do |certname|
+            if CERTNAME_BLACKLIST.include?(certname)
+              errors << "    Cannot manage cert named `#{certname}` from " +
+                        "the CLI, if needed use the HTTP API directly"
+            end
+
+            unless certname =~ VALID_CERTNAME
+              errors << '  Certname %{name} must not contain unprintable or non-ASCII characters'
+            end
+          end
+        end
+
+        errors_were_handled = Utils.handle_errors(@logger, errors, parser.help)
+
+        exit_code = errors_were_handled ? 1 : nil
+
+        return results, exit_code
+      end
+
+      def run(input)
+        certnames = input['certnames']
+        config_path = input['config']
+
+        # Validate config_path provided
+        if config_path
+          errors = Puppetserver::Utils::FileUtilities.validate_file_paths(config_path)
```

Comment:
Since you include `Puppetserver::Utils` at the top of this class you should be able to just reference `FileUtilities` here and below.

Reply:
Thanks!  I'll make sure to change all of them in the file